File: EShipment/Models/Order.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace EShipment.Models
{
    public class Order
    {
      public Order()
      {
      }

    public long ID { get; set; }

      [ForeignKey("ApplicationUser")]
      public String ApplicationUser_Id { get; set; }

      public virtual ApplicationUser ApplicationUser { get; set; }

      public string Number { get; set; }

      public string Mark { get; set; }

      public string ContainerNumber { get; set; }

      public string Destination { get; set; }

      public string DischargedPort { get; set; }

      public string BLNumber { get; set; }

      public DateTime? ETD { get; set; }

      public DateTime? ETA { get; set; }

      public int? NumbOfGoods { get; set; }

      public string Weight { get; set; }

      public string Measurement { get; set; }

      public string ProductDescription { get; set; }

      private IList<OrderStatus> statuses = new List<OrderStatus>();
      public virtual IList<OrderStatus> Statuses { get { return statuses; } set { statuses = value; } }

      public DateTime? ReceiveOrderDate { get; set; }
  }
}

File: EShipment/client-src/app/model/order.ts
import { OrderStatus } from './orderStatus';

export class Order {
  id: number;
  applicationUser_Id: string;
  number: string;
  companyName: string;
  mark: string;
  containerNumber: string;
  destination: string;
  dischargedPort: string;
  blNumber: string;
  etd: Date;
  eta: Date;
  numOfGoods: number;
  weight: number;
  measurement: number;
  productDescription: string;
  statuses: OrderStatus[];
  receiveOrderDate: Date;
  constructor();
  constructor(number: string, companyName: string, mark: string, containerNumber: string, destination: string,
    dischargedPort: string, blNumber: string, etd: Date, ETA: Date, numOfGoods: number, weight: number,
    measurement: number, productDescription: string, statuses: OrderStatus[], receiveOrderDate: Date);
  constructor(number?: string, companyName?: string, mark?: string, containerNumber?: string, destination?: string,
    dischargedPort?: string, blNumber?: string, etd?: Date, eta?: Date, numOfGoods?: number, weight?: number,
    measurement?: number, productDescription?: string, statuses?: OrderStatus[], receiveOrderDate?: Date) {
    this.number = number;
    this.companyName = companyName;
    this.mark = mark;
    this.containerNumber = containerNumber;

    this.destination = destination;
    this.dischargedPort = dischargedPort;
    this.blNumber = blNumber;
    this.etd = etd;

    this.eta = eta;
    this.numOfGoods = numOfGoods;
    this.weight = weight;
    this.measurement = measurement;

    this.productDescription = productDescription;
    this.statuses = statuses;
    this.receiveOrderDate = receiveOrderDate;
  }
}
